### DISCUSSION TITLE
[Roland TD-50][ASIO] - In- & Output Channels are fixed

### DISCUSSION THREAD

**Discussion by danbeut** [jamulussoftware/jamulus#1216] [Technical Support & Help]
Hi community,

I'm playing a Roland TD-50 V-Drum and want to follow Rule #1 to only listen to the Server Signal and not my Modules signal.
I used JamKazam in the past for a short period and it allowed me to configure the input independently from the output which also allows me to use a headset and communicate with my bandmates via the server.

I'm now wondering why I can't change the output channel to a headphone / headset which is connected via USB or 3.5

Does anyone have an idea or a hint how I can follow rule #1 with my Roland TD-50? (Sub is also TD-50 related and not the server signal from the computer directly)

![Screenshot 2021-03-10 105237](https://user-images.githubusercontent.com/80388448/110610940-02126b00-818f-11eb-95cb-ab58c67cec8e.png)

**Comment by ann0see** [comment]
I think in theory you could either open two Jamulus instances and listen to the one where you set the correct outputs or use ASIO4ALL (which I wouldn't recommend). I think this limitation is due to the way ASIO works. Currently Jamulus only supports one ASIO driver. On a Mac your suggested workflow should be possible easily.

**Reply by danbeut** [reply]
Thank you for your answer! 
But as I don't have a Mac and have not planned to buy such I'll try with two instances of Jamulus.

Why wouldn't it be possible on a Mac but not with any other Computer?

**Reply by ann0see** [reply]
Because of ASIO which (except for ASIO4ALL) only supports one device. Maybe once PortAudio (another way of internally handling audio processing) works well, this might also work on windows. 

The limitation sounds stupid, but probably is due to synchronisation problems which might occur if you connect multiple devices.

**Comment by KonFey** [comment]
If you are looking for a software only solution:- have you tried a SW mixer like Voicemeeter Banana - it allows you to select different input and output drivers. So it should be possible to configure the virtual ASIO in Jamulus, and get the input from TD-50 and reroute the output to ASIO4ALL.
Note however that this step adds additional processing to your computer and will -depending on your device -  add delay and jitter to your audio signal. I experimented a bit with this solution but did not persue it further. 

My preferred way to connect my Roland drum set (TD-9) is to use the analog output of TD-9, connect it to a mixer with analog input and USB output (Yamaha AG 06 in my case, but any other device recommended on the Jamulus page should work) and hook up my headset there as well. 
It brings me the additional benefit that I can mix in a microphone as well - much more fun if you can talk to your co-musicians as well... or does your TD-50 have a mic in channel?
